src/feedback.ts

```diff
@@ -0,0 +1,196 @@
+    repository(owner:$owner, name:$name){
+      pullRequests(last:100, labels: $labels){
+        nodes {
+          id, number, updatedAt
+          labels(first: 100) {
+            nodes { id, name }
+          }
+          timelineItems(last: 100, itemTypes: [
+            LABELED_EVENT, BASE_REF_CHANGED_EVENT, CLOSED_EVENT, ISSUE_COMMENT, RENAMED_TITLE_EVENT,
+            PULL_REQUEST_COMMIT, PULL_REQUEST_REVIEW, READY_FOR_REVIEW_EVENT, REVIEW_REQUESTED_EVENT
+          ]) {
+            nodes {
+              __typename
+              ... on LabeledEvent {
+                createdAt
+                label { name }
+                actor { login }
+              }
+              ... on ClosedEvent { actor { login } }
+              ... on IssueComment { author { login } }
+              ... on RenamedTitleEvent { actor { login } }
+              ... on PullRequestCommit { pullRequest { author { login } } }
+              ... on PullRequestReview { author { login } }
+              ... on ReadyForReviewEvent { actor { login } }
+              ... on ReviewRequestedEvent { actor { login } }
+            }
+          }
+        }
+      }
+    }
+}`
+
+const commentMutationPart = `
+  addComment(input: {subjectId: $itemId, body: $body}) {
+    clientMutationId
+  }
+`
+
+const addLabelsMutationPart = `
+  addLabelsToLabelable(input: {labelableId: $itemId, labelIds: [$closeLabelId]}){
+    clientMutationId
+  }
+`
+
+const removeLabelsMutationPart = `
+  removeLabelsFromLabelable(input: {labelableId: $itemId, labelIds: $labelIds}){
+    clientMutationId
+  }
+`
+
+const closeIssueMutation = `mutation ($itemId: ID!, $body: String!, $labelIds: [ID!]!, $closeLabelId: ID!) {
+  ${commentMutationPart}
+  closeIssue(input: {issueId: $itemId, stateReason: NOT_PLANNED}) {
+    clientMutationId
+  }
+  ${addLabelsMutationPart}
+  ${removeLabelsMutationPart}
+}`
+
+const closePullMutation = `mutation ($itemId: ID!, $body: String!, $labelIds: [ID!]!, $closeLabelId: ID!) {
+  ${commentMutationPart}
+  closePullRequest(input: {pullRequestId: $itemId}) {
+    clientMutationId
+  }
+  ${addLabelsMutationPart}
+  ${removeLabelsMutationPart}
+}`
+
+const removeLabelsMutation = `mutation ($itemId: ID!, $labelIds: [ID!]!) {
+   ${removeLabelsMutationPart}
+ }`
+
+function queryParams(context: Context, feedbackLabels: config.FeedbackLabels): RequestParameters {
+  return {
+    owner: context.repo.owner,
+    name: context.repo.repo,
+    labels: Array.from(feedbackLabels.keys())
+  }
+}
+
+function wasUpdatedAfterLabeling(events: any[]): boolean {
+  const authors = new Set()
+  var requester = null
+  for (let i = events.length - 1; i >= 0; i--) {
+    const event = events[i]
+    if (event.__typename !== 'LabeledEvent') {
+      const author = event.actor?.login ?? event.author?.login ?? event.pullRequest?.author?.login
+      authors.add(author)
+    } else if (event.label.name.startsWith('pending:')) {
+      requester = event.actor.login
```

Comment:
🤔 I assume the intention here is that if I add the `pending:*` label before I post the message, the label should stay.

The implementation will never remove a pending label if the person adding it later (hours later, days later) adds a message.

Should we add a time cutoff for this? So it only catches the accidental removal where labeling and commenting happened out of order?

Reply:
You are correct.
I thought about this, but I don't think this use case occurs often enough to care about it. In any case, we can add it later if its a problem.